### PR TITLE
[DSM] PEPPER-706 AT export fix

### PR DESCRIPTION
**Pull request by pegahtah**
## Context

_Say a few words about the bigger picture beyond this PR. How do your changes improve life for our users? Link directly to the jira ticket (just type "DDP-..." and it'll automatically link) to help your teammates quickly get up to speed on the motivation for these changes._

## Checklist

- [ ] I have labeled the type of changes involved using the `C-*` labels.
- [ ] I have assessed potential risks and labeled using the `R-*` labels.
- [ ] I have considered error handling and alerts, and added `L-*` labels as needed.
- [ ] I have considered security and privacy, and added `I-*` labels as needed
- [ ] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ ] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ ] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [x] I have written automated positive tests
- [ ] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/migration/DynamicFieldsMappingMigrator.java [hunk 1]
```diff
@@ -45,6 +45,7 @@ public void export() {
         for (FieldSettingsDto fieldSettingsDto : fieldSettingsByStudyName) {
             parser.setDisplayType(fieldSettingsDto.getDisplayType());
             parser.setPossibleValuesJson(fieldSettingsDto.getPossibleValues());
+            parser.setFieldName(fieldSettingsDto.getColumnName());
```
Comment (pegahtah) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/migration/DynamicFieldsMappingMigrator.java:48]: This was throwing NPE

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/util/ElasticSearchUtil.java [hunk 2]
```diff
@@ -516,6 +518,7 @@ public static void writeWorkflow(RestHighLevelClient client, @NonNull WorkflowFo
             } else {
                 workflowMapES = addWorkflows(workflow, status, workflowForES.getStudySpecificData());
             }
+            logger.info("WorkflowMapES: "+ workflowMapES);
```
Comment (pegahtah) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/util/ElasticSearchUtil.java:521]: Adding extra logs to see AT workflow updates

File: pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/elasticExport/ElasticSearchExportTest.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+@Slf4j
+public class ElasticSearchExportTest {
+    @BeforeClass
+    public static void first(){
+        TestHelper.setupDB();
+    }
+
+    @Test
+    public void dsmeExportParticipantListIsMigrationTrueSpecificStudy(){
```
Comment (pegahtah) [inline pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/elasticExport/ElasticSearchExportTest.java:22]: Added a unit test to run the export per study without the need for pubsub

File: pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/elasticExport/ElasticSearchExportTest.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            exportPayload.setIsMigration(IS_MIGRATION);
+            try {
+                DSMtasksSubscription.migrateToES(exportPayload);
+            } catch (Exception e) {
+                error = true;
+                log.error(study + " had the following error while export: ");
+                e.printStackTrace();
+            }
+            log.info("----------------DSM export finished for "+ study+" ---------------------");
+        }
```
Comment (pegahtah) [inline pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/elasticExport/ElasticSearchExportTest.java:56]: Added a positive test to make sure all the exports run without any errors

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/painless/UpsertPainless.java [hunk 5]
```diff
@@ -40,6 +40,8 @@ public void export() {
         updateByQueryRequest.setScript(painless);
         updateByQueryRequest.setMaxRetries(5);
         updateByQueryRequest.setRefresh(true);
+        updateByQueryRequest.setAbortOnVersionConflict(false);
```
Comment (pegahtah) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/painless/UpsertPainless.java:43]: @ssettipalli your change is here

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/util/ElasticSearchUtil.java (hunk 2 above)
Comment (otchet-broad) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/util/ElasticSearchUtil.java:521]: Will this ever have PHI or PII in it?

File: pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/elasticExport/ElasticSearchExportTest.java (hunk 4 above)
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/elasticExport/ElasticSearchExportTest.java:56]: Can you please trigger the CircleCI tests?  I suspect this test might fail there.

**Review by ssettipalli — APPROVED** [review]
logger statements could use some cleanup and add in the format like
logger.info("WorkflowMapES:  {} ", workflowMapES);
looks ok.

File: pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/elasticExport/ElasticSearchExportTest.java (hunk 4 above)
Comment (ssettipalli) [reply pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/elasticExport/ElasticSearchExportTest.java:56]: @pegahtah 
Do we really want to run this test which triggers ALL study full dsm exports each time test suite is run ?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/painless/UpsertPainless.java [hunk 6]
```diff
@@ -40,6 +40,8 @@ public void export() {
...
+        logger.info("Export isAbortOnVersionConflict: " + updateByQueryRequest.isAbortOnVersionConflict());
```
Comment (ssettipalli) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/painless/UpsertPainless.java:44]: we can remove this log statement

**Pull request merged**